src/shared/components/pipeline-run-logs/logs/Logs.tsx

```diff
@@ -41,21 +42,37 @@ const Logs: React.FC<LogsProps> = ({
   const contentRef = React.useRef<HTMLDivElement>(null);
   const [error, setError] = React.useState<boolean>(false);
   const resourceStatusRef = React.useRef<string>(resourceStatus);
+  const blockContentRef = React.useRef<string>('');
+
   const onCompleteRef = React.useRef<(name) => void>();
   onCompleteRef.current = onComplete;
 
   const appendMessage = React.useRef<(blockContent) => void>();
 
+  const addContent = debounce(() => {
+    contentRef.current.innerText += blockContentRef.current;
+    blockContentRef.current = '';
+  }, 200);
```

Comment:
This should be wrapped as a `React.useCallback`. However once that's done, i suspect you can run into issues where updates continue to happen within the 200ms and therefore the screen never gets updated until we get a large chunk. I suggest throttling instead of debouncing here.

Reply:
Throttling dint work, it is causing too many scrollIntoView calls and browser couldn't handle that. However debounce may take very little extra time to update the screen but eventually when it does, there will be one scroll operation that takes us to the bottom of the page, users can scroll up to view content without any issue and also resume the stream back.